### ISSUE TITLE
Accessing an array in the expression it was created sends the accessor to the abyss as an array

### ISSUE THREAD

**Issue by thennothinghappened** [YoYoGames/GameMaker-Bugs#4508]
### Description
As in the very convoluted title, if you try to access an array as you create it (see below), the compiler sees the array access as another array declaration, and instead just returns the original array. These can also be chained indefinitely, to do nothing of value, if you feel like it!

### Steps To Reproduce
```gml
var a = [][0];

// "[  ]"
show_debug_message(a);
```
The contents of `a` are set to the initial array, and the accessor becomes an array itself. I had a look at the HTML5 codegen for this and it just creates the accessor as an array:
```js
function gml_GlobalScript_test_script(_inst,_other)
{
    var gmla=__yy_gml_array_create([]);
    __yy_gml_array_create([0]);
    show_debug_message(gmla);
}
```

You can even go further:
```gml
var a = [][0]["test"][][][][][][];

// "[  ]"
show_debug_message(a);
```
And it'll just create more empty arrays and forget about them.

[44111d1a-d11e-49db-a3b9-352e3cacd6b8](https://api.gamemaker.io/api/github/downloads/44111d1a-d11e-49db-a3b9-352e3cacd6b8)

**Comment by thennothinghappened** [comment]
Slight further testing:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/35243139/97a191b9-fe19-4d6d-b5ec-7ff331668aaf)

**Comment by rwkay** [comment]
Fixed in 2024.2 - this is now fixed across all VM and YYC - but I cannot see a legitimate reason for doing this, at least syntactically it will work

**Comment by cameron-home** [comment]
Empty arrays are no longer returned in IDE v2024.200.0.486  Runtime v2024.200.0.502, so verifying as fixed now.
